cuda_core/tests/test_program.py

```diff
@@ -38,14 +36,44 @@ def test_program_with_various_options(init_cuda):
         ProgramOptions(diag_error=1234, diag_suppress=1234),
         ProgramOptions(diag_error=[1234, 1223], diag_suppress=(1234, 1223)),
         ProgramOptions(diag_warn=1000),
+    ],
+)
+def test_cpp_program_with_various_options(init_cuda, options):
+    code = 'extern "C" __global__ void my_kernel() {}'
+    program = Program(code, "c++", options)
+    assert program.backend == "nvrtc"
+    program.compile("ptx")
+    program.close()
+    assert program.handle is None
+
+
+options = [
+    ProgramOptions(max_register_count=32),
+    ProgramOptions(debug=True),
+    ProgramOptions(lineinfo=True),
+    ProgramOptions(ftz=True),
+    ProgramOptions(prec_div=True),
+    ProgramOptions(prec_sqrt=True),
+    ProgramOptions(fma=True),
+]
+if not _linker._decide_nvjitlink_or_driver():
+    print("Using nvjitlink as the backend because decide() returned false")
+    options += [
+        ProgramOptions(time=True),
+        ProgramOptions(split_compile=True),
     ]
 
-    for options in options_list:
-        program = Program(code, "c++", options)
-        assert program.backend == "nvrtc"
-        program.compile("ptx")
-        program.close()
-        assert program.handle is None
+
+@pytest.mark.parametrize("options", options)
+def test_ptx_program_with_various_options(init_cuda, options):
+    code = 'extern "C" __global__ void my_kernel() {}'
+    program = Program(code, "c++")
+    ptx_object_code = program.compile("ptx")
```

Comment:
We could consider caching the generated PTX (it's the same for all parameters if test speed is a concern.